Week3/homework/src/deleteToDoItem.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+function deleteToDoItem(connection, errorHandler, req, res) {
+  const username = req.params.username;
+  const password = req.params.password;
+  const listDescription = req.params.listDescription;
+  const toDoDescription = req.params.toDoDescription;
+  const queryString1 = 'SELECT user_id FROM users WHERE user_name = ? AND user_password = ?';
+  connection.query(queryString1, [username, password], (err, rows1) => {
+    errorHandler(err, res);
+    const queryString2 = 'SELECT list_id FROM lists WHERE user_id = ? AND description = ?;';
+    connection.query(queryString2, [rows1[0].user_id, listDescription], (err, rows2) => {
+      errorHandler(err, res);
+      const queryString3 = 'DELETE FROM items WHERE list_id = ? AND description = ?';
```

Comment:
**Non-blocking comment:**

There is a good practice to use LIMIT in your queries: https://dba.stackexchange.com/questions/52079/does-using-limit-improve-the-performance-and-is-it-noticeable

Long story short: 
- in case of select of only one row, you want to return the result once the row is found because you're looking for only one row, and it would be wise to stop the search when the one is found. However, if you select by PK - SQL server will know it and will stop the search itself - because of PK is unique
- in case of deletion - it's good for security reasons, and also saves you from occasional deleting of multiple rows. 

Look at your query right know: if you've got 3 records with the same description in the same list - all 3 of them will be deleted.

Reply:
I've changed this URI and connected it to item_id as PK as well.